react-common/components/controls/Input.tsx

```diff
@@ -83,7 +85,11 @@ export const Input = (props: InputProps) => {
     }
 
     const changeHandler = (e: React.ChangeEvent<any>) => {
-        const newValue = (e.target as any).value;
+        let newValue = (e.target as any).value;
+        if (newValue && filter) {
+            const pat = new RegExp(filter);
```

Comment:
I'd suggest moving this RegExp to a state variable so it isn't recreated on each change.

Reply:
Moved it. Does it look okay?